File: mbw/src/main/java/com/mycelium/wallet/external/changelly/ChangellyTransactionOffer.java
package com.mycelium.wallet.external.changelly;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

public class ChangellyTransactionOffer implements Serializable {
    public String id;
    public double apiExtraFee;
    public double changellyFee;
    public String payinExtraId;
    public String status;
    public String currencyFrom;
    public String currencyTo;
    public double amountTo;
    public double amountFrom; // not set by changelly (!) but we set it on the way back in ChangellyService
    public String payinAddress;
    public String payoutAddress;
    public String createdAt;

}

File: mbw/src/main/java/com/mycelium/wallet/external/changelly/model/ChangellyTransactionOffer.kt
package com.mycelium.wallet.external.changelly.model

import java.io.Serializable
import java.math.BigDecimal


class ChangellyTransactionOffer : Serializable {
    @JvmField
    var id: String? = null
    @JvmField
    var payinExtraId: String? = null
    var status: String? = null
    @JvmField
    var currencyFrom: String? = null
    var currencyTo: String? = null
    @JvmField
    var amountTo:BigDecimal = BigDecimal.ZERO
    @JvmField
    var payinAddress: String? = null
    var payoutAddress: String? = null
    var payoutExtraId: String? = null
    var createdAt: String? = null

    val amountExpectedFrom:BigDecimal = BigDecimal.ZERO

    var trackUrl: String? = null
    var type: String? = null
    var refundAddress: String? = null
    var refundExtraId: String? = null
}